### DISCUSSION TITLE
Using with file-based sqlite

### DISCUSSION THREAD

**Discussion by thefrisson** [sinaptik-ai/pandas-ai#650] [General]
The library documentation suggests that connecting to a sqlite database should be easy using SQLConnector (as that's what is given in the example), but I am looking to connect a simple file-based sqlite db, not a server hosted version. I tried to pass the SQLConnector config a filename and it pretty much said all of the parameters that are in the config in the example are required. But it seems like the logical progression for constructing something like this is to work on supporting file based databases first and then do mysql and postgres. I understand that there is a propensity to overlook and oversimplify examples when there's is a lot of functionality to show. So I am wondering if anyone knows of a way to configure the SQLConnector to accept a file that doesn't have a host, port, username, or password?

**Comment by gventuri** [comment]
@thefrisson thanks a lot for reporting, it's definitely a connector we'll need to add.
